nestkernel/archiving_node.cpp

```diff
@@ -228,6 +312,126 @@ nest::Archiving_Node::clear_history()
   Kminus_ = 0.0;
   triplet_Kminus_ = 0.0;
   history_.clear();
+  Ca_minus_ = 0.0;
+  Ca_t_ = 0.0;
+}
+
+
+/* ----------------------------------------------------------------
+* Get the number of synaptic_elements
+* ---------------------------------------------------------------- */
+double_t
+nest::Archiving_Node::get_synaptic_element( Name n ) const
+{
+  std::map< Name, SynapticElement >::const_iterator se_it;
+  se_it = synaptic_elements_map_.find( n );
+  double_t z_value;
+
+  if ( se_it != synaptic_elements_map_.end() )
+  {
+    z_value = ( se_it->second ).get_z();
+    if ( ( se_it->second ).continuous() )
+    {
+      return z_value;
+    }
+    else
+      return std::floor( z_value );
+  }
+  else
+  {
+    return 0.0;
+  }
+}
+
+int_t
+nest::Archiving_Node::get_synaptic_element_vacant( Name n ) const
+{
+  std::map< Name, SynapticElement >::const_iterator se_it;
+  se_it = synaptic_elements_map_.find( n );
+
+  if ( se_it != synaptic_elements_map_.end() )
+  {
+    return se_it->second.get_z_vacant();
+  }
+  else
+  {
+    return 0;
+  }
+}
+
+int_t
+nest::Archiving_Node::get_synaptic_element_connected( Name n ) const
+{
+  std::map< Name, SynapticElement >::const_iterator se_it;
+  se_it = synaptic_elements_map_.find( n );
+
+  if ( se_it != synaptic_elements_map_.end() )
+  {
+    return se_it->second.get_z_connected();
+  }
+  else
+  {
+    return 0;
+  }
+}
+
+std::map< Name, double_t >
+nest::Archiving_Node::get_synaptic_elements() const
+{
+  std::map< Name, double_t > n_map;
+
+  for ( std::map< Name, SynapticElement >::const_iterator it = synaptic_elements_map_.begin();
+        it != synaptic_elements_map_.end();
+        ++it )
+  {
+    n_map.insert( std::pair< Name, double_t >( it->first, get_synaptic_element( it->first ) ) );
+  }
+  return n_map;
+}
+
+void
+nest::Archiving_Node::update_synaptic_element( double_t t )
+{
+  if ( t < Ca_t_ )
```

Comment:
Could this happen in reality, or is this only a safeguard against errors elsewhere in the code? In the latter case, turn this into `assert( t >= Ca_t);`